test/migration/migration_test.go

```diff
@@ -0,0 +1,219 @@
+package migration
+
+import (
+	"flag"
+	"log"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/coreos/prometheus-operator/pkg/client/monitoring/v1alpha1"
+	"github.com/coreos/prometheus-operator/pkg/k8sutil"
+	operatorFramework "github.com/coreos/prometheus-operator/test/framework"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var framework *operatorFramework.Framework
+var kubeconfig, opImage, ns, ip *string
+
+func TestMain(m *testing.M) {
+	kubeconfig = flag.String("kubeconfig", "", "kube config path, e.g. $HOME/.kube/config")
+	opImage = flag.String("operator-image", "", "operator image, e.g. quay.io/coreos/prometheus-operator")
+	ns = flag.String("namespace", "prometheus-operator-e2e-tests", "e2e test namespace")
+	flag.Parse()
+
+	os.Exit(m.Run())
+}
+
+func TestMigration(t *testing.T) {
+	// Create all the clients required.
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kclient, err := kubernetes.NewForConfig(config)
```

Comment:
@Gouthamve Ah, OK. Can you add a comment above the line so I remember in the future?